datadog/resource_datadog_synthetics_test_.go

```diff
@@ -1106,14 +1110,14 @@ func updateSyntheticsTestLocalState(d *schema.ResourceData, syntheticsTest *data
 		key := clientCertificate.GetKey()
 		localCertificate["key"][0]["filename"] = key.GetFilename()
 
-		// the content of the certificate and the key are write-only
-		// so we need to get them from the config since they will
-		// not be in the api response
+		// the content of client certificate is write-only so it will not be returned by the API.
+		// To avoid useless diff but also prevent storing the value in clear in the state
+		// we store a hash of the value.
 		if configCertificateContent, ok := d.GetOk("request_client_certificate.0.cert.0.content"); ok {
-			localCertificate["cert"][0]["content"] = configCertificateContent.(string)
+			localCertificate["cert"][0]["content"] = getCertificateStateValue(configCertificateContent.(string))
```

Comment:
I think `GetOk` takes the value from the config, not the state, so it should never be hashed already. Unless I'm missing something, there should be no need to do this, and instead rely on the `StateFunc` you added above to let terraform store in the state no ?

Reply:
I thought so as well, but when I print the value to debug, we get a hashed value (even after commenting the `StateFunc`). I assumed it is when terraform reads the config to compare the state and the config.

Reply:
Yeah looks like depending on when you call it, it picks from state or from config...